app/lib/differential_expression_service.rb

```diff
@@ -303,6 +306,18 @@ def self.validate_study(study)
     raise ArgumentError, "#{study.accession} cannot view cluster plots" unless study.can_visualize_clusters?
   end
 
+  # determine if a study has author-uploaded DE results
+  # this will stop the automatic calculation of new DE results if they add more data
+  #
+  # * *params*
+  #   - +study+ (Study) => Study to validate
+  #
+  # * *returns*
+  #   - (Boolean)
+  def self.study_has_author_de?(study)
+    study.study_files.by_type('Differential Expression').any?
```

Comment:
So would this prevent any further automatic calculation even in cases where they had kept the SCP DE results and just added more files after?